Rename test file time.chpl to avoid ambiguity with Time.chpl

bradcray opened the pull request:
This is to avoid ambiguities on case-insensitive platforms.

bradcray commented:
With this change, I get clean testing on Mac in the library/packages/TOML test directory.  I'd feel better if someone who knows Mason / TOML better (@spartee, @ben-albrecht) were to review and merge on my behalf rather than doing it myself though.

ben-albrecht commented:
Confirmed that this testing error was a result of #13930.

Prior to #13930, one could use module names that collided with standard library without warning.

bradcray commented:
I've dug into this a bit and think I can explain what's going on.  Nothing seems obviously wrong even though the change in behavior was surprising:

I'm reasonably certain that what's happening is that, prior to #13930, since many of our modules were referring to the contents of `Sort` without actually `use`-ing it, they were benefitting from other modules' `use` statements to resolve it.  Since #13930 inserted `use` statements into every module that referred to its contents, the parsing order for modules like `Sort` (and `Time`, which it `use`s) changed, reflecting the order in which these new `use` statements were encountered.

The other part of this is that we currently resolve `use` statements for internal modules differently than for package modules and user code.  Specifically, the latter consider modules from arbitrary locations while the former only permit well-known modules to be considered to avoid surprises/hijacking of things the internal modules were built to assume they'd have available.  So adding these new `use` statements also caused certain cases to start considering user modules as candidates for resolving them.  In this specific case, since `Sort` is a package module and it uses `Time`, candidates in both the user path and the standard paths are considered.  In this case, since there was a file `time.chpl` in the test directory, it was considered as a candidate on case-insensitive platforms.

I think this compiler logic (and these rules) could definitely benefit from being  reconsidered, cleaned up, and/or reimplemented, but I think what's happening is consistent with what we've implemented and is just a side effect of having inserted more `use` statements than we had before.  Maybe put another way, if the same `use Sort` statements are put onto master prior to #13930 with no other changes, the same warning takes place.
